pkg/apis/core/validation/validation.go

```diff
@@ -6541,6 +6542,22 @@ func ValidateMaxSkew(fldPath *field.Path, maxSkew int32) *field.Error {
 	return nil
 }
 
+// validateMinDomains tests that the argument is a valid MinDomains.
+func validateMinDomains(fldPath *field.Path, minDomains *int32, action core.UnsatisfiableConstraintAction) field.ErrorList {
+	if minDomains == nil {
+		return nil
+	}
+	var allErrs field.ErrorList
+	if *minDomains <= 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, minDomains, isNotPositiveErrorMsg))
+	}
+	// When MinDomains is non-nil, whenUnsatisfiable must be DoNotSchedule.
+	if action != core.DoNotSchedule {
```

Comment:
In `SetDefaults_Pod`, `MinDomains` will default to 1. We should check both `action` and `MinDomains` value here, right?

Reply:
We just changed the default to be `nil` (that is, we don't default anything). This validation is correctly matching the defined API.

Reply:
Ah, okay. I forgot to remove the code for MinDomains default value in `SetDefaults_Pod`. Thanks.